### DISCUSSION TITLE
Documentation not really to understand - getting only temperature_2m

### DISCUSSION THREAD

**Discussion by lucabert** [open-meteo/open-meteo#966] [Q&A]
Hi all!

I got the server working in a docker container, created **not** using the delivered Docker-files, but starting from an Ubuntu 22.04 image.

I can query the server and get the current temperature at 2 meter.
Unfortunately, I cannot get any other value.

I configured my `/etc/default/openmeteo-api.env` as follow:
```
# API service configuration
VAPOR_ENV=production
LOG_LEVEL=info
API_BIND="0.0.0.0:8080"
API_APIKEYS_PATH=
API_SYNC_APIKEYS=

# Email address to send failure notifications to
# NOTIFICATION_EMAIL=

# Sync client configuration
SYNC_ENABLED=true
SYNC_APIKEY=
SYNC_SERVER=
SYNC_DOMAINS=cmc_gem_gdps,dwd_icon
SYNC_VARIABLES=temperature_2m,dew_point_2m,visibility,wind_speed_10m,wind_direction_10m,cloud_cover_1000hPa,cloud_cover_975hPa,cloud_cover_950hPa,cloud_cover_925hPa,cloud_cover_900hPa,cloud_cover_850hPa,cloud_cover_800hPa,cloud_cover_700hPa
SYNC_PAST_DAYS=7
SYNC_CONCURRENT=4
SYNC_REPEAT_INTERVAL=10
```

But when I try to get data, only `temperature_2m` is delivered:

```bash
curl "http://127.0.0.1:8080/v1/forecast?latitude=51.134333333333&longitude=13.768&current=temperature_2m,relative_humidity_2m,precipitation,rain,showers,snowfall,weather_code,pressure_msl,wind_speed_10m,wind_direction_10m&wind_speed_unit=kn&timezone=Europe%2FBerlin" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   680    0   680    0     0   221k      0 --:--:-- --:--:-- --:--:--  221k
{
  "latitude": 51.14,
  "longitude": 13.759998,
  "generationtime_ms": 0.4730224609375,
  "utc_offset_seconds": 7200,
  "timezone": "Europe/Berlin",
  "timezone_abbreviation": "CEST",
  "current_units": {
    "time": "iso8601",
    "interval": "seconds",
    "temperature_2m": "°C",
    "relative_humidity_2m": "%",
    "precipitation": "mm",
    "rain": "mm",
    "showers": "mm",
    "snowfall": "cm",
    "weather_code": "wmo code",
    "pressure_msl": "hPa",
    "wind_speed_10m": "kn",
    "wind_direction_10m": "°"
  },
  "current": {
    "time": "2024-08-20T15:00",
    "interval": 900,
    "temperature_2m": 26.5,
    "relative_humidity_2m": null,
    "precipitation": null,
    "rain": null,
    "showers": null,
    "snowfall": null,
    "weather_code": null,
    "pressure_msl": null,
    "wind_speed_10m": null,
    "wind_direction_10m": null
  }
}
```
Can someone explain me what I'm doing wrong?
I compiled the codes using `docs/development.md`.

**Comment by patrick-zippenfenig** [comment]
Hi,
weather variables are often computed on demand. E.g. wind speed is stored in U/V components. You will have to get `wind_u_component_10m, wind_v_component_10m `. You can get a complete list here: https://openmeteo.s3.amazonaws.com/index.html#data/dwd_icon/

Currently there is no documentation available, how weather variables are derived. But you can get a list from the source code: https://github.com/open-meteo/open-meteo/blob/9bc8c8d5571a108ca4bb7db4128386a4506ad141/Sources/App/Icon/IconReader.swift#L282.

**Comment by delphir-com** [comment]
@patrick-zippenfenig is there a way to set `SYNC_DOMAINS` to "all" instead of listing each and every value?

**Reply by patrick-zippenfenig** [reply]
No. It is unlikely that users really need all models und would generate tremendous amount of traffic for the open-data distribution.

**Comment by delphir-com** [comment]
the reason I ask is this commit:
[945669f5a6bbde4f9d22bf8465919202d8332cbc](https://github.com/open-meteo/open-meteo/commit/945669f5a6bbde4f9d22bf8465919202d8332cbc) which removed ncep_gefs025_probability support and silently broke syncing service with this update... so I`m thinking how to prevent such issues in the future

**Reply by patrick-zippenfenig** [reply]
Got it! In the future, I’ll "deactivate" retired domains and add a warning. Moving forward, we aim to adopt a cloud-native strategy to access data on demand, rather than pre-downloading everything. Given the high traffic on AWS, I suspect that only a small portion of this data is actually used.

**Reply by delphir-com** [reply]
another option which could save the traffic: store data in S3 in a way which would allow you to download forecasts for specific days (dates) only. so that if you are interested in `now to +3 days` data only - then you don't need to download anything before/after this range. and even if you need a timezone adjustment - it is still just +/-1 day added to the range what is still less than "everything"
